Gifski/SavePanelAccessoryViewController.swift

```diff
@@ -1,86 +1,257 @@
 import Cocoa
-import AVFoundation
+import AVKit
 
 final class SavePanelAccessoryViewController: NSViewController {
+	enum PredefinedSizeItem {
+		case custom
+		case spacer
+		case dimensions(ResizableDimensions)
+
+		var resizableDimensions: ResizableDimensions? {
+			switch self {
+			case let .dimensions(resizableDimensions):
+				return resizableDimensions
+			default:
+				return nil
+			}
+		}
+	}
+
 	@IBOutlet private var estimatedSizeLabel: NSTextField!
-	@IBOutlet private var scaleSlider: NSSlider!
-	@IBOutlet private var scaleLabel: NSTextField!
 	@IBOutlet private var frameRateSlider: NSSlider!
 	@IBOutlet private var frameRateLabel: NSTextField!
 	@IBOutlet private var qualitySlider: NSSlider!
+
+	@IBOutlet private var widthTextField: IntTextField!
+	@IBOutlet private var heightTextField: IntTextField!
+	@IBOutlet private var predefinedSizesDropdown: MenuPopUpButton!
+	@IBOutlet private var dimensionsTypeDropdown: MenuPopUpButton!
+
 	var inputUrl: URL!
 	var videoMetadata: AVURLAsset.VideoMetadata!
 	var onDimensionChange: ((CGSize) -> Void)?
 	var onFramerateChange: ((Int) -> Void)?
 
+	let formatter = ByteCountFormatter()
+
+	private var resizableDimensions: ResizableDimensions!
+	private var predefinedSizes: [PredefinedSizeItem]!
+
 	override func viewDidLoad() {
 		super.viewDidLoad()
 
-		let formatter = ByteCountFormatter()
 		formatter.zeroPadsFractionDigits = true
+		setupDimensions()
+		setupDropdowns()
+		setupSliders()
+		setupWidthAndHeightTextFields()
+	}
+
+	override func viewWillAppear() {
+		super.viewWillAppear()
+
+		// Hack to enlarge extended save panel frame
+		UserDefaults.standard.set("{841, 681}", forKey: "NSNavPanelExpandedSizeForSaveMode")
```

Comment:
added 200pts in height from what we had before

Reply:
Can you add the original initial size in a comment, so it's clear how much it changes when reading the code?